### DISCUSSION TITLE
Nested sub-routes question

### DISCUSSION THREAD

**Discussion by kczx3** [leafsphp/leaf#186] [Q&A] [answered]
Can you provide an explanation of the valid routes that the second example on the [Sub-routing documentation page](https://leafphp.dev/docs/routing/sub-routing.html#sub-routing) would provide?  I fear I am not understanding the intended effect of it.  At the time of this writing, below is the example code I am referring to.  I have added comments mimicking the first example with my understanding and what actually happens.

```php
$app->group('/user', function () use($app) {
  // will result in '/user/'
  $app->get('/', function () use($app) {
    $app->response()->markup('no user id');
  });

  // will result in '/user/12'
  $app->get('/(\d+)', function ($id) use($app) {
    $app->response()->markup("user $id");
  });

  // I would have thought this would correspond to '/user/settings'
  // However, this actually mounts onto the root path '/settings/'
  $app->mount('/settings', function () use($app) {
    // will result in '/settings/privacy' when I expected '/user/settings/privacy'
    $app->get('/privacy', function () use($app) {
      $app->response()->markup('Privacy Settings');
    });

    // will result in '/settings/notification' when I expected '/user/settings/notification'
    $app->get('/notification', function () use($app) {
      $app->response()->markup("Notification Settings");
    });
  });
});
```

**Comment by kczx3** [comment]
If this works as intended and not how I imagine then perhaps it would be nice to pass the current route path into the callback function that you pass into `$app->group()`.
```php
$app->group('/user', function () use($app, $routePrefix1) {
  // $routePrefix1 would be equal to '/user' here
  // I would have thought this would correspond to '/user/settings'
  // However, this actually mounts onto the root path '/settings/'
  $app->mount("$routePrefix1/settings", function () use($app, $routePrefix2) {
    // $routePrefix2 would be equal to '/user/settings' here however in this case it is not needed
    // will result in '/settings/privacy' when I expected '/user/settings/privacy'
    $app->get('/privacy', function () use($app) {
      $app->response()->markup('Privacy Settings');
    });
  });
});
```

**Reply by mychidarko** [reply]
@kczx3 I thought I had replied to this thread. This is actually a bug, the behaviour you expected is the right one.

**Reply by mychidarko** [reply, marked as answer]
It's been fixed now, so you can update Leaf and your issue should be fixed

**Reply by kczx3** [reply]
Thanks!  You [replied on Discord](https://discord.com/channels/891808595715436574/891812439388196884/1095771228666609754) but I think there was some misunderstanding.  The wording in the docs about nested routes not supporting dynamic url patterns is confusing.  You might consider re-wording it or providing an example of what _doesn't work_ in addition to the existing sample of what you _can do_.

**Reply by kczx3** [reply]
@mychidarko I'm not sure that this is resolved 100%.  Nested group routes only work for the first sub-group.  Take the following example:

```php
$app->group('/test', function() {
    app()->get('/', fn() => response()->plain('test root'));

    app()->group('/one', function () {
        app()->get('/', fn() => response()->plain('test/one'));
    });

    app()->group('/two', function () {
        app()->get('/', fn() => response()->plain('test/two'));
    });
});
```

After this, `app()->routes()` returns the following:
![image](https://user-images.githubusercontent.com/15206020/235214722-8d9d83b3-6419-4a2a-ad3a-2525bad2bb31.png)

Shouldn't the last route be mounted onto `/test/two`?

**Reply by mychidarko** [reply]
Let me check this out. Might be some funny logic

**Reply by mychidarko** [reply]
It's been fixed @kczx3

**Reply by kczx3** [reply]
Working now.  Thanks!!!
